client/src/routes/Temples/components/DateTimePicker.tsx

```diff
@@ -0,0 +1,170 @@
+import RNDateTimePicker from '@react-native-community/datetimepicker';
+import dayjs from 'dayjs';
+import React, {Dispatch, SetStateAction, useEffect, useState} from 'react';
+import {Modal, Platform} from 'react-native';
+import styled from 'styled-components/native';
+import utc from 'dayjs/plugin/utc';
+
+import {COLORS} from '../../../common/constants/colors';
+import {SPACINGS} from '../../../common/constants/spacings';
+
+import Button from '../../../common/components/Buttons/Button';
+import {TopSafeArea} from '../../../common/components/Spacers/Spacer';
+import TouchableOpacity from '../../../common/components/TouchableOpacity/TouchableOpacity';
+import {
+  Body16,
+  BodyBold,
+} from '../../../common/components/Typography/Body/Body';
+import {useTranslation} from 'react-i18next';
+import NS from '../../../lib/i18n/constants/namespaces';
+
+dayjs.extend(utc);
+
+const Wrapper = styled.View({
+  backgroundColor: COLORS.WHITE,
+  borderRadius: SPACINGS.EIGHT,
+});
+
+const Row = styled(TouchableOpacity)({
+  paddingHorizontal: SPACINGS.SIXTEEN,
+  paddingVertical: SPACINGS.EIGHT,
+  flexDirection: 'row',
+  justifyContent: 'space-between',
+});
+
+const ModalBackground = styled.View({
+  justifyContent: 'center',
+  backgroundColor: COLORS.BLACK_TRANSPARENT,
+  flex: 1,
+});
+
+const ModalView = styled.View({
+  margin: SPACINGS.TWENTY,
+  backgroundColor: COLORS.WHITE,
+  borderRadius: SPACINGS.TWELVE,
+  padding: SPACINGS.THIRTYTWO,
+  alignItems: 'center',
+  shadowColor: '#000',
+  shadowOpacity: 0.25,
+  shadowRadius: 4,
+  elevation: 5,
+});
+
+const DoneButton = styled(Button)({alignSelf: 'center'});
+
+const DateTimePicker: React.FC<{
+  mode: 'date' | 'time';
+  selectedValue: dayjs.Dayjs;
+  setValue: Dispatch<SetStateAction<dayjs.Dayjs>>;
+  close: () => void;
+  maximumDate?: Date;
```

Comment:
Maybe expose these as Dayjs and convert so we hide that the underlying component needs Date

Reply:
Yessss! that's da way for sure, I'll update it! 🔥